digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -110,7 +117,9 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
     //pure linkId based queries, maybe something related to the zoomLevel we are in map level.
     val fetchMissingRoadAddressStartTime = System.currentTimeMillis()
     val (floatingViiteRoadLinks, addresses, floating) = Await.result(fetchRoadAddressesByBoundingBoxF, Duration.Inf)
-    val missingLinkIds = linkIds -- floating.keySet -- addresses.keySet
+    val complementedWithChangeAddresses = applyChanges(complementedRoadLinks, filteredChangedRoadLinks, addresses)
+    //TODO sub complementedWithChangeAddresses to addresses
```

Comment:
We create the list of missing link ids by taking the list of all link ids, removing the link ids that have floating road addresses, non-floating addresses, and post-transfer addresses.

This method should be refactored when speed improvements are developed in Sprint 2.